qiskit_nature/circuit/library/initial_states/fermionic_gaussian_state.py

```diff
@@ -0,0 +1,121 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Fermionic Gaussian states."""
+
+from typing import Optional, Sequence
+
+import numpy as np
+
+from qiskit import QuantumRegister, QuantumCircuit
+
+from qiskit_nature.converters.second_quantization import QubitConverter
+from qiskit_nature.mappers.second_quantization import JordanWignerMapper
+
+from ._helper import _prepare_fermionic_gaussian_state_jordan_wigner
+
+
+class FermionicGaussianState(QuantumCircuit):
+    """A circuit that prepares a fermionic Gaussian state."""
+
+    def __init__(
+        self,
+        transformation_matrix: np.ndarray,
+        occupied_orbitals: Optional[Sequence[int]] = None,
+        qubit_converter: QubitConverter = None,
+        **circuit_kwargs,
+    ) -> None:
+        r"""Initialize a circuit that prepares a fermionic Gaussian state.
+
+        A fermionic Gaussian state is a state of the form
+
+        .. math::
+            b^\dagger_1 \cdots b^\dagger_{N_p} \lvert \overline{\text{vac}} \rangle
+
+        where
+
+        .. math::
+           \begin{pmatrix}
+                b^\dagger_1 \\
+                \vdots \\
+                b^\dagger_N \\
+           \end{pmatrix}
+           = W
+           \begin{pmatrix}
+                a^\dagger_1 \\
+                \vdots \\
+                a^\dagger_N \\
+                a_1 \\
+                \vdots \\
+                a_N
+           \end{pmatrix},
+
+        - :math:`a^\dagger_1, \ldots, a^\dagger_{N}` are the fermionic creation operators
+        - :math:`W` is an :math:`N \times 2N` matrix such that
+          :math:`b^\dagger_1, \ldots, b^\dagger_{N}` also satisfy the
+          fermionic anticommutation relations
+        - :math:`\lvert \overline{\text{vac}} \rangle` is the mutual 0-eigenvector of
+          the operators :math:`\{b_j^\dagger b_j\}`
+
+        The matrix :math:`W` has the block form
+
+        .. math::
+           \begin{pmatrix}
+                W_1 & W_2
+           \end{pmatrix}
+
+        where :math:`W_1` and :math:`W_2` must satisfy
+
+        .. math::
+            W_1 W_1^\dagger + W_2 W_2^\dagger = I \\
+            W_1 W_2^T + W_2 W_1^T = 0
+
+        Currently, only the Jordan-Wigner Transformation is supported.
+
+        Reference: arXiv:1711.05395
+
+        Args:
+            transformation_matrix: The matrix :math:`W` that specifies the coefficients of the
+                new creation operators in terms of the original creation and annihilation operators.
+                This matrix must satisfy special constraints; see the docstring of this function.
+            qubit_converter: a QubitConverter instance.
```

Comment:
`occupied_orbitals` is missing from the documentation. This should also explain why the it will fall back to zero occupied orbitals and what effect this has

Reply:
Done.